src/v/resource_mgmt/cpu_profiler.cc

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "resource_mgmt/cpu_profiler.h"
+
+#include "random/generators.h"
+#include "resource_mgmt/logger.h"
+#include "ssx/future-util.h"
+#include "ssx/sformat.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/internal/cpu_profiler.hh>
+#include <seastar/core/reactor.hh>
+
+#include <iterator>
+
+namespace resources {
+
+cpu_profiler::cpu_profiler(
+  config::binding<bool>&& enabled,
+  config::binding<std::chrono::milliseconds>&& sample_rate)
+  : _query_timer([this] { return poll_samples(); })
```

Comment:
drop return since poll_samples return void

Reply:
Good catch, dropped the return.